jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryEndpointProvider.java

```diff
@@ -53,15 +54,15 @@
   T handleResponse(Response response) throws IOException, RegistryException;
 
   /**
-   * Handles an {@link HttpResponseException} that occurs.
+   * Handles an {@link ResponseException} that occurs.
    *
-   * @param responseException the {@link HttpResponseException} to handle
+   * @param responseException the {@link ResponseException} to handle
    * @throws HttpResponseException {@code responseException} if {@code responseException} could not
    *     be handled
    * @throws RegistryErrorException if there is an error with a remote registry
    */
-  default T handleHttpResponseException(HttpResponseException responseException)
-      throws HttpResponseException, RegistryErrorException {
+  default T handleHttpResponseException(ResponseException responseException)
```

Comment:
Not sure if we should have `default` if we fully control the workflow here... RegistryEndpointProvider is internal. Can we file a followup to cleanup this part?

Reply:
Filed #2117.